bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/internal/base/conditions/mining/LocalConditions.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.internal.base.conditions.mining;
+
+import java.nio.file.Path;
+import java.util.Collection;
+
+import org.eclipse.passage.lic.internal.api.LicensedProduct;
+import org.eclipse.passage.lic.internal.api.LicensingException;
+import org.eclipse.passage.lic.internal.api.conditions.Condition;
+import org.eclipse.passage.lic.internal.api.conditions.mining.ConditionMiningException;
+import org.eclipse.passage.lic.internal.api.conditions.mining.MinedConditions;
+import org.eclipse.passage.lic.internal.api.io.KeyKeeperRegistry;
+import org.eclipse.passage.lic.internal.api.io.StreamCodecRegistry;
+import org.eclipse.passage.lic.internal.api.registry.StringServiceId;
+import org.eclipse.passage.lic.internal.base.i18n.BaseMessages;
+import org.eclipse.passage.lic.internal.base.io.FileCollection;
+import org.eclipse.passage.lic.internal.base.io.PassageFileExtension;
+import org.eclipse.passage.lic.internal.base.io.PathFromLicensedProduct;
+
+@SuppressWarnings("restriction")
+public abstract class LocalConditions implements MinedConditions {
+
+	protected final StringServiceId id;
+	private final KeyKeeperRegistry keys;
+	private final StreamCodecRegistry codecs;
+
+	protected LocalConditions(StringServiceId id, KeyKeeperRegistry keys, StreamCodecRegistry codecs) {
+		this.id = id;
+		this.keys = keys;
+		this.codecs = codecs;
+	}
+
+	@Override
+	public final StringServiceId id() {
+		return id;
+	}
+
+	@Override
+	public final Collection<Condition> all(LicensedProduct product) throws ConditionMiningException {
+		return conditions(licenses(product));
+	}
+
+	private Collection<Path> licenses(LicensedProduct product) throws ConditionMiningException {
+		try {
+			return new FileCollection(new PathFromLicensedProduct(this::base, product),
+					new PassageFileExtension.LicenseEncrypted()).get();
+		} catch (LicensingException e) {
+			throw new ConditionMiningException(//
+					String.format(BaseMessages.getString("PathConditionMiner.failure"), id, product), e); //$NON-NLS-1$
+		}
+	}
+
+	private Collection<Condition> conditions(Collection<Path> sources) throws ConditionMiningException {
+		return null;
```

Comment:
OMG, I can't believe my eyes!

Reply:
It's just started - draft class with autogenerated interface method implementation. The following  PR is going to contain sode here.